fvm/src/machine/default.rs

```diff
@@ -117,12 +120,23 @@ where
         #[cfg(not(any(test, feature = "testing")))]
         engine.preload(state_tree.store(), builtin_actors.left_values())?;
 
+        let randomness = externs.get_chain_randomness(
+            0,
+            context.epoch,
```

Comment:
It is unsupported behaviour to call `get_chain_randomness` with an epoch greater than or equal to the current epoch (the `context.epoch` in this case).

If it's not too much work, I would suggest looking for a simpler source of randomness here.